map_gen/shared/generate.lua

```diff
@@ -282,7 +282,9 @@ end
 function Public.init(args)
     tiles_per_tick = args.tiles_per_tick or 32
     regen_decoratives = args.regen_decoratives or false
-    surfaces = args.surfaces or {}
+    for surface_name, shape in pairs(args.surfaces) do
```

Comment:
args.surfaces could be nil
```suggestion
    for surface_name, shape in pairs(args.surfaces or {}) do
```